libraries/chain/worker_proposal_evaluators.cpp

```diff
@@ -58,4 +59,76 @@ namespace golos { namespace chain {
+            "Techspec can be created only under proposal post");
+        GOLOS_CHECK_LOGIC(wpo_itr->state == created,
+            logic_exception::this_worker_proposal_already_has_approved_techspec,
+            "This worker proposal already has approved techspec");
+        GOLOS_CHECK_LOGIC(wpo_itr->type == task,
+            logic_exception::this_worker_proposal_has_premade_work,
+            "This worker proposal has premade work");
+
+        const auto& wto_idx = _db.get_index<worker_techspec_index, by_permlink>();
+        auto wto_itr = wto_idx.find(std::make_tuple(o.author, o.permlink));
+        if (wto_itr != wto_idx.end()) {
+            GOLOS_CHECK_LOGIC(o.specification_cost.symbol == wto_itr->specification_cost.symbol,
+                logic_exception::cannot_change_cost_symbol,
+                "Cannot change cost symbol");
+            GOLOS_CHECK_LOGIC(o.development_cost.symbol == wto_itr->development_cost.symbol,
+                logic_exception::cannot_change_cost_symbol,
+                "Cannot change cost symbol");
+
+            _db.modify(*wto_itr, [&](worker_techspec_object& wto) {
+                wto.modified = now;
+                wto.specification_cost = o.specification_cost;
+                wto.specification_eta = o.specification_eta;
+                wto.development_cost = o.development_cost;
+                wto.development_eta = o.development_eta;
+                wto.payments_count = o.payments_count;
+                wto.payments_interval = o.payments_interval;
+            });
+
+            return;
+        }
+
+        _db.create<worker_techspec_object>([&](worker_techspec_object& wto) {
+            wto.author = o.author;
+            wto.permlink = comment.permlink;
+            wto.created = now;
+            wto.specification_cost = o.specification_cost;
+            wto.specification_eta = o.specification_eta;
+            wto.development_cost = o.development_cost;
+            wto.development_eta = o.development_eta;
+            wto.payments_count = o.payments_count;
+            wto.payments_interval = o.payments_interval;
+        });
+    }
+
+    void worker_techspec_delete_evaluator::do_apply(const worker_techspec_delete_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_20__1013, "worker_techspec_delete_operation");
+
+        const auto& wto_idx = _db.get_index<worker_techspec_index, by_permlink>();
+        auto wto_itr = wto_idx.find(std::make_tuple(o.author, o.permlink));
+        if (wto_itr == wto_idx.end()) {
+            GOLOS_THROW_MISSING_OBJECT("worker_techspec_object", fc::mutable_variant_object()("author",o.author)("permlink",o.permlink));
+        }
+
+        // TODO: check wpo has not this techspec as approved
+        //const auto& wpo_idx = _db.get_index<worker_proposal_index, by_permlink>();
+        //auto wpo_itr = wpo_idx.find(std::make_tuple(comment.parent_author, comment.parent_permlink));
```

Comment:
You shouldn't add limits on delete tspec.

You should unfreeze asset for approved tspec if author want to delete it.